pages/api/user/index.js

```diff
@@ -0,0 +1,49 @@
+import User from "@/db/models/user";
+import dbConnect from "@/db/connect";
+import { getServerSession } from "next-auth/next";
+
+export default async function handler(req, res) {
+  const session = await getServerSession(req, res);
+
+  if (session) {
+    await dbConnect();
+
+    const { user } = session;
+    if (req.method === "GET") {
+      const currentUser = await User.findOne(user).populate("roasts");
+
+      if (!currentUser) {
+        return res.status(404).json({ error: "Something bad happened" });
+      }
+      return res.status(201).json(currentUser);
+    }
+
+    if (req.method === "POST") {
+      const id = req.body;
+      const currentUser = await User.findOne(user);
+      if (!currentUser) {
+        return res.status(404).json({ error: "Something bad happened" });
+      }
```

Comment:
Really good! You're checking if the currently logged-in user still exists in the database. Theoretically, it could happen that a user is deleted from the database by someone else, and in that case they shouldn't be able to access the database anymore.